### ISSUE TITLE
Apexchart not drawn first time if component is blended out with v-show

### ISSUE THREAD

**Issue by DennisBergemann** [apexcharts/vue-apexcharts#395]
Hi,

in my case i have a footermenu that blends in the components respective to the menu with v-show.
Now for the component with apexcharts in it, the charts are, for the firsttime i swap to the menupoint, not drawn.
Only if a swap the menu and then back its drawn. 

Or if i make the component with the chart as the first (shown) then it is drawn too

For all other components it works fine

you can test like this...

`<v-main>
      <UserComponent v-show="navItem == 0" />
      <StatisticsComponent v-show="navItem == 1" />
    </v-main>

    <v-bottom-navigation
      app
      background-color="primary lighten1"
      dark
      color="yellow"
      :value="navItem"
    >
      <v-btn @click="navItem = 0"
        ><span>Users</span><v-icon>mdi-account</v-icon></v-btn
      >
      
      <v-btn @click="navItem = 1"
        ><span>Statistics</span><v-icon>mdi-trending-up</v-icon></v-btn
      >
    </v-bottom-navigation>`

set the navItem in data to 0 as first selection. Now when you start you see the UserComponent and when you swap in menu the StatisticsComponent but without Charts...

Only if you click back to UserComponent and back to StatisticsComponent the charts are drawn as aspected.

Thank you for reading and i hope you can investigate for an solution

Thanks a lot

**Comment by wdporter** [comment]
same issue me. Did you find a workaround ?

**Comment by DennisBergemann** [comment]
Yes i solved it for me. I tested different ways. Some of them were real crap and put a strain on my reload.
But then I found a good solution that works in a resource-saving way.

First add a ref to the component that contains your charts like this:
`<v-main>
      <VdiDevices v-if="user" v-show="navItem == 1" />
      <VdiUserSessions v-if="user" v-show="navItem == 0" />
      <VdiUserErrors v-if="user" v-show="navItem == 2" />
      <VdiStatistics v-if="user" v-show="navItem == 3" ref="vdiStatistics" />
    </v-main>`

then add in methods for this file:
`sendRefresh() {
      log.Info("sending refresh");
      this.$refs.vdiStatistics.RefreshCharts();
    }`
    
In your component which contains the charts add the following method to do the redraw magic with resize
`async RefreshCharts() {
      window.dispatchEvent(new Event("resize"));
    },`
    
    


then for your bottom nav call the method when changing to the view with the chart component like this
`<v-bottom-navigation
      app
      background-color="primary lighten1"
      dark
      color="yellow"
      :value="navItem"
    >
      <v-btn @click="navItem = 0"
        ><span>Users</span><v-icon>mdi-account</v-icon></v-btn
      >
      <v-btn @click="navItem = 1"
        ><span>Devices</span><v-icon>mdi-devices</v-icon></v-btn
      >

      <v-btn @click="navItem = 2"
        ><span>User Logs</span><v-icon>mdi-account</v-icon></v-btn
      >
      <v-btn
        @click="
          navItem = 3;
          sendRefresh();
        "
        ><span>HostPool Statistics</span><v-icon>mdi-trending-up</v-icon></v-btn
      >
    </v-bottom-navigation>`
    
    
In Summary:
1. in the chart component call the resize event
2. in the app where is your menu create an ref to the component with the charts and call with this ref the resize event


I don't think I can describe it any better right now. Give it a try and get back to me. I'd love to hear if it worked.
